mitodl/edx-sga
# Serialize and parse solution as an XML element, if valid XML

#### What are the relevant tickets?
Fixes #210 

#### What's this PR do?
Adds parsing and rendering of solution elements as XML elements instead of the XML attribute. When parsing the attribute will still be used if the element also doesn't exist. When rendering it will render as an XML element if the solution is valid XML, else it will store it in the attribute.

#### How should this be manually tested?
 - Create a `.tar.gz` file from the new `test_data/2017_SGA` directory. The files and directories in that directory should all be at the root level of the `.tar.gz` file. For example:

        cd test_data/2017_SGA
        tar czvf ../pr-211.tar.gz *

 - In studio create a new edX course. Then go to Tools -> Import and import your `.tar.gz` file. You should see a successful upload
 - Go to the SGA unit in studio and click 'Edit'. You should see `You're seeing the answer` in the textbox at the very bottom.
 - Go to Tools -> Export. Export a tarball. Untar it unto a directory and look at `course/vertical/vertical.xml`. It should look like this:

        <vertical display_name="SGA Unit">
          <edx_sga url_name="edx_sga" xblock-family="xblock.v1" display_name="SGA Test 1">
            <solution>
              <p>You're seeing the answer</p>
            </solution>
          </edx_sga>
        </vertical>
